File: player.js
import * as THREE from "https://cdn.skypack.dev/three";

export class Player {
  constructor(scene) {
    this.health = 100;
    this.mesh = new THREE.Mesh(
      new THREE.BoxGeometry(1, 2, 1),
      new THREE.MeshStandardMaterial({ color: 0x0000ff })
    );
    this.mesh.position.set(0, 10, 0);
    scene.add(this.mesh);
  }

  update(zone) {
    // Gravity
    if (this.mesh.position.y > 1) this.mesh.position.y -= 0.1;

    // Zone damage
    if (!zone.isInside(this.mesh.position)) {
      this.health -= 0.05;
      document.getElementById("health").innerText =
        `Health: ${Math.max(0, this.health.toFixed(0))}`;
    }
  }
}

File: player.py
class Player:
    def __init__(self):
        self.x = 640
        self.y = 360
        self.health = 100
        self.radius = 10

    def update(self, zone):
        if not zone.is_inside(self.x, self.y):
            self.health -= 0.05

    def draw(self, screen):
        import pygame
        pygame.draw.circle(screen, (0, 0, 255), (int(self.x), int(self.y)), self.radius)
